### PR TITLE
[timeseries] drop statsmodels, improve refit_every_n_windows

### PR DESCRIPTION
**Pull request by canerturkmen**
*Issue #, if available:*

#4062 

*Description of changes:*

Two housekeeping items
- Drop unused `statsmodels` dependency
- Improve docs and raise warning for `refit_every_n_windows`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

File: timeseries/src/autogluon/timeseries/predictor.py [hunk 1]
```diff
@@ -631,7 +631,11 @@ def fit(
             This argument has no effect if ``tuning_data`` is provided.
         refit_every_n_windows: int or None, default = 1
             When performing cross validation, each model will be retrained every ``refit_every_n_windows`` validation
-            windows. If set to ``None``, model will only be fit once for the first validation window.
+            windows, where the number of validation windows is specified by `num_val_windows`. Note that in the
+            default setting where `num_val_windows=1`, this argument has no effect.
+
+            If set to ``None``, model will only be fit once for the first validation window. By default,
+            `refit_every_n_windows=`, i.e., models will be refit for all validation windows.
```
Comment (shchur) [inline timeseries/src/autogluon/timeseries/predictor.py:638]:
```suggestion
            `refit_every_n_windows=1`, i.e., all models will be refit for each validation windows.
```

File: timeseries/src/autogluon/timeseries/predictor.py [hunk 2]
```diff
@@ -631,7 +631,11 @@ def fit(
             This argument has no effect if ``tuning_data`` is provided.
         refit_every_n_windows: int or None, default = 1
             When performing cross validation, each model will be retrained every ``refit_every_n_windows`` validation
-            windows. If set to ``None``, model will only be fit once for the first validation window.
+            windows, where the number of validation windows is specified by `num_val_windows`. Note that in the
+            default setting where `num_val_windows=1`, this argument has no effect.
+
+            If set to ``None``, model will only be fit once for the first validation window. By default,
```
Comment (shchur) [inline timeseries/src/autogluon/timeseries/predictor.py:637]:
```suggestion
            If set to ``None``, models will only be fit once for the first (oldest) validation window. By default,
```

**Review by shchur — APPROVED** [review]
Nice, thank you!

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Pull request merged**